src/fileio.cpp

```diff
@@ -527,10 +527,23 @@ FILE *FioFOpenFile(const char *filename, const char *mode, Subdirectory subdir,
 
 /**
  * Create a directory with the given name
+ * If the parent directory does not exist, it will try to create that as well.
  * @param name the new name of the directory
  */
 void FioCreateDirectory(const char *name)
 {
+	char dirname[MAX_PATH];
+	strecpy(dirname, name, lastof(dirname));
```

Comment:
Why choose `strecpy` over `strcpy`? What is the difference between the two?

Reply:
strecpy is an OTTD-special that has a end-point parameter. Helps prevent buffer overruns (e.g. if name is longer than MAX_PATH)